<!-- The setting of the network has to be done through tkDNN -->
<!-- This node only uses the provided engine -->

<launch>
  <arg name="image_src" default="/image_raw"/>
  <arg name="camera_id" default="/cam02"/>

  <arg name="pretrained_model_file" default="$(find vision_tkdnn_detect)/serialized_models/yolo3_fp16.rt"/>
  <arg name="score_threshold" default="0.30"/>
  <arg name="num_batches" default="1"/>
  <arg name="num_classes" default="80"/>

  <node pkg="vision_tkdnn_detect" name="vision_tkdnn_detect" type="vision_tkdnn_detect" output="screen" required="true">
    <param name="image_raw_node" type="str" value="$(arg camera_id)$(arg image_src)"/>

    <param name="pretrained_model_file" type="str" value="$(arg pretrained_model_file)"/>
    <param name="score_threshold" type="double" value="$(arg score_threshold)"/>
    <param name="num_batches" type="int" value="$(arg num_batches)"/>
    <param name="num_classes" type="int" value="$(arg num_classes)"/>
  </node>

  <node pkg="detected_objects_visualizer" type="visualize_rects" name="yolo3_rects"
        output="screen">
    <param name="image_src" value="$(arg camera_id)$(arg image_src)"/>
    <param name="image_out" value="/image_rects"/>
    <param name="object_src" value="/detection/image_detector/objects"/>
  </node>


</launch>
